engine/tests/ElggEntityTest.php

```diff
@@ -262,4 +262,26 @@ public function testElggEntityArrayAppendMetadata() {
 			$this->assertEqual(array_merge($md, $md2), $obj->$name);
 		}
 	}
+
+	public function testElggEntityGetIconURL() {
+
+		function get_icon_url_test_hook($hook, $type, $url, $params) {
```

Comment:
Could you make this an anon function? I know PHP allows this, but, ick, global function creation.